src/modules/hip/kernel/non_silent_region_detection.hpp

```diff
@@ -0,0 +1,406 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 0 -  moving mean square kernel device helpers --------------------
+
+__host__ __device__ int smem_pos(int pos)
+{
+    return pos + (pos >> 5);
```

Comment:
Why is it >> 5? Is it since you are considering 32 banks? That could be hardware specific and will change with arch. Pls add a comment in any case.